### PR TITLE
Remove C++ templates, because they are stupid

### PR DESCRIPTION
**Pull request by alexey-milovidov**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-ch-test-poll3** [comment]
*This is an automated comment for commit 660d3df3721cba0026f67dde7cbf1f9cd43d0851 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/55910/660d3df3721cba0026f67dde7cbf1f9cd43d0851/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/6600971899">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/55910/660d3df3721cba0026f67dde7cbf1f9cd43d0851/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: src/Functions/currentProfiles.cpp [hunk 1]
```diff
@@ -17,39 +17,46 @@ namespace
...
+    {
+        switch (kind)
+        {
+            case Kind::currentProfiles: return "currentProfiles";
+            case Kind::enabledProfiles: return "enabledProfiles";
+            case Kind::defaultProfiles: return "defaultProfiles";
+        }
+    }
+
     class FunctionCurrentProfiles : public IFunction
```
Comment (SmitaRKulkarni) [inline src/Functions/currentProfiles.cpp:35]: Maybe we can call it `FunctionProfiles`

**Review by SmitaRKulkarni — COMMENTED** [review]
Rest all LGTM

File: src/Functions/currentProfiles.cpp [hunk 2]
```diff
@@ -17,39 +17,46 @@ namespace
 {
     enum class Kind
     {
-        CURRENT_PROFILES,
-        ENABLED_PROFILES,
-        DEFAULT_PROFILES,
+        currentProfiles,
+        enabledProfiles,
+        defaultProfiles,
```
Comment (vitlibar) [inline src/Functions/currentProfiles.cpp:22]: [14. For the constants in an enum, use CamelCase with a capital letter. ALL_CAPS is also acceptable. If the enum is non-local, use an enum class.](https://clickhouse.com/docs/en/development/style#how-to-write-code)

Comment (alexey-milovidov) [reply src/Functions/currentProfiles.cpp:22]: Let's break the rule...

**Pull request merged**